src/gui_common/CustomCheckBox.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   A customized check box that changes icon when hovered / clicked.
+/// </summary>
+public class CustomCheckBox : Button
+{
+    private Texture unpressedNormal;
+    private Texture unpressedHovered;
+    private Texture unpressedClicked;
+    private Texture pressedNormal;
+    private Texture pressedHovered;
+    private Texture pressedClicked;
+
+    private bool pressing;
+    private CheckState lastCheckState;
+    private CheckState currentCheckState;
+
+    private enum CheckState
+    {
+        UnpressedNormal,
+        UnpressedHovered,
+        UnpressedClicked,
+        PressedNormal,
+        PressedHovered,
+        PressedClicked,
+        Disabled,
+    }
+
+    public override void _Ready()
+    {
+        unpressedNormal = GetIcon("UnpressedNormal", "CheckBox");
+        unpressedHovered = GetIcon("UnpressedHovered", "CheckBox");
+        unpressedClicked = GetIcon("UnpressedClicked", "CheckBox");
+        pressedNormal = GetIcon("PressedNormal", "CheckBox");
+        pressedHovered = GetIcon("PressedHovered", "CheckBox");
+        pressedClicked = GetIcon("PressedClicked", "CheckBox");
+        UpdateIcon();
+    }
+
+    public override void _Process(float delta)
```

Comment:
Turns out there's actually a way to properly update the icons in `_Draw` method. You just need to defer the custom icon update call so it doesn't conflict with the internal one. Can be done by using `Invoke.Queue` method.

Reply:
What was the problem regarding that when _Draw is just overridden and looks like this:
```
protected override void _Draw(){
base._Draw();
// do our custom stuff here
```
to override what happens by default?

Reply:
There's no problem with the `_Draw` override itself, the problem is when setting `Icon` property in the `_Draw` method. So what happens is:
1. mouse hovers button area,
2. button redraws (which includes text and icon) and `draw` notification is propagated,
4. we set `Icon` with custom hovered icon in `_Draw`
5. `Icon` is set thus `draw` is propagated once more

All this happens concurrently and the two icon redraw call conflicts with each other so by deferring the `UpdateIcon` call we kind of separates the initial hovered redraw and the subsequent custom icon redraw.